lib/linter-views.coffee

```diff
@@ -33,30 +33,52 @@ class LinterViews
       isTextEditor = paneItem?.getPath?
       @bottomContainer.setVisibility(isTextEditor)
       @panel.panelVisibility = atom.config.get('linter.showErrorPanel') and isTextEditor
-    @subscriptions.add @linter.onDidChangeMessages =>
-      @render()
+      @render(@linter.messages.publicMessages)
     @subscriptions.add @bottomContainer.onDidChangeTab =>
       @renderPanelMessages()
 
-  render: ->
-    @messages = @linter.messages.getAllMessages()
+  render: (messages) ->
+    @messages = @classifyMessagesByLine(@classifyMessages(messages))
```

Comment:
Do you ever call `classifyMessagesByLine` or `classifyMessages` separate from each other? If not, then one should call the other.